common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -290,6 +291,26 @@ def student_view(self, context):
                 banner_text, special_html = special_html_view
                 if special_html and not masquerading_as_specific_student:
                     return Fragment(special_html)
+
+        # Content type gating for FBE previously only gated individual blocks
+        # This was an issue because audit learners could start a timed exam and then be unable to complete the exam
+        # even if they later upgrade because the timer would have expired.
+        # For this reason we check if content type gating is enabled for the user and gate the entire sequence in that event
+        if self.is_time_limited and self.location.block_type == 'sequential':
+            try:
+                user = User.objects.get(id=self.runtime.user_id)
+                # importing here to avoid a circular import
+                from openedx.features.content_type_gating.models import ContentTypeGatingConfig
```

Comment:
Rather than having this code live in the sequence module, it might be better to add knowledge to the block-transformer that currently actively gates graded problems. That way you wouldn't be dealing w/ this import cycle, and would instead be just adding knowledge of another kind of feature to be gated.

Reply:
Even if we're leaving this code here, I think it's worth a note about the future plans with the MFE to explain why this level of pulling in openedx features is warranted. (This breaks the ability to use this sequence block outside of `edx-platform`, basically. I think we're ok w/ that, but it's worth a note).

@ormsbee, can you double check me on this?

Reply:
If we wanted to make it less likely to break outside of the platform, we could add another XBlock service that has access to content type gating information.

Reply:
Agreed on all points.

Reply:
@cpennington so should I do what you're suggesting? if so could you elaborate?

Reply:
Leave a note about this breaking compatibility w/ using sequence outside edx-platform, but don't do any further eng work on it.